[quant][graphmode] quantization support for aten::rehshape

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33650 [not4land] debugging
* #33533 [not4land] Temporary workaround to get quantized ops
* #34855 [quant][graphmode] quantization support for prim::CallFunction
* #34346 [quant][graphmode] Add quantization support for aten::cat
* #34345 [quant][graphmode] Add prim::ListConstruct to general op handling
* #34854 [quant][graphmode] quantization support for view, transpose, contiguos
* #34806 [quant][graphmode] quantization support for aten::chunk
* #34807 [quant][graphmode] quantization support for prim::ListUnpack
* #34804 [quant][graphmode] Replicate quantize node for prim::If
* #34572 [quant][graphmode] quantization support for aten::add
* #34518 [quant][graphmode] quantization work for prim::If
* #34571 [quant][graphmode][fix] use observed_values_ to check values are observed
* #34414 [quant][graphmode][refactor] insertObservers for Block
* #34411 [quant][graphmode][fix] Insert dequantize before use node
* #34349 [quant][graphmode][fix] preserve the type of original value when inserting dequant node
* #34348 [quant] Add dequantize.tensors
* **#34803 [quant][graphmode] quantization support for aten::rehshape**
* #34347 [quant][graphmode] Add quantization support for aten::dropout

Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504457](https://our.internmc.facebook.com/intern/diff/D20504457)

jerryzh168 committed:
[quant][graphmode] quantization support for aten::rehshape

Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::rehshape"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::rehshape"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::rehshape"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::rehshape"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar reviewed (commented):
The test added just seems to check if dequantize will be swapped. Don't we need a test to check reshape op standalone?

jerryzh168 commented:
@supriyar making sure the dequantize will be swapped after reshape is how we quantize it.

jerryzh168 commented:
For context, please see: https://github.com/pytorch/pytorch/commit/776d2a1e8f37cfb7619c682c5af488d3c883d087#diff-1aea24a8684a93a8be65d804f891ee85

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::rehshape"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504457](https://our.internmc.facebook.com/intern/diff/D20504457)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] quantization support for aten::rehshape"


Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20504457](https://our.internmc.facebook.com/intern/diff/D20504457)

[ghstack-poisoned]
